hydra-redis/test/querying_test.py

```diff
@@ -0,0 +1,127 @@
+        else:
+            return False
+
+    def classendpointTest(self):
+        """Test for class endpoint"""
+        query = "show classEndpoints"
+        data = query_facades.user_query(query)
+        if len(data) == 2:
+            return True
+        else:
+            return False
+
+    def collectionendpointTest(self):
+        """Test for collection endpoint"""
+        query = "show collectionEndpoints"
+        data = query_facades.user_query(query)
+        if len(data) == 9:
+            return True
+        else:
+            return False
+
+    def CommandCollectionmemberTest(self):
+        """
+        Test for all Commands in CommandCollection.
+        Data is already stored in check_data from the static data url.
+        Check_data is used for compare the data retrieve by querying process.
+        """
+        check_data = ['[]']
+        query = "show CommandCollection members"
+        data = query_facades.user_query(query)
+        if data[1] == check_data:
+            return True
+        else:
+            return False
+
+    def ControllerLogCollectionmemberTest(self):
+        """
+        Test for all controller logs for ControllerLogCollection.
+        Whole object of ControllerLogCollection is stored in check data.
+        Check_data is used for compare the data retrieve by querying process.
+        """
+        check_data = [{'@id': '/api/ControllerLogCollection/65',
+                       '@type': 'ControllerLog'},
+                      {'@id': '/api/ControllerLogCollection/183',
+                       '@type': 'ControllerLog'},
+                      {'@id': '/api/ControllerLogCollection/374',
+                       '@type': 'ControllerLog'}]
+        query = "show ControllerLogCollection members"
+        data = query_facades.user_query(query)
+        # Make data searchable and comaprable.
+        data1 = str(data[1]).replace('"', '')
+        # data retrive from the memory can be distributed:
+        # like type can be at first position and id can be at second.
+        # So, check_data split in 3 parts.
+        # And check all parts are in data retrieve.
+        if str(
+            check_data[0]) in data1 and str(
+            check_data[1]) in data1 and str(
+                check_data[2]) in data1:
+            return True
+        else:
+            return False
+
+    def DatastreamCollectionmemberTest(self):
+        """Test for all datastream with Drone ID 2"""
+        check_data = ['/api/DatastreamCollection/19']
+        query = "show DatastreamCollection members"
+        data = query_facades.user_query(query)
+        # Here are find the datastream only for those which have DroneID 2.
+        query = "show DroneID 2 and type Datastream"
+        data = query_facades.user_query(query)
+        if data == check_data:
+            return True
+        else:
+            return False
+
+
+class TestQueryingMechanism(unittest.TestCase):
+    query_test = Tests()
```

Comment:
Also, use teardown and setup methods for testing.

Reply:
I can not figure out, what should I put in setUp and tearDown function? I think there are the different things in every test because of the different query in every test nothing is similar to put in setUp function.